app/containers/MessageBox/index.js

```diff
@@ -292,17 +292,16 @@ class MessageBox extends Component {
 				const { start, end } = this.component._lastNativeSelection;
 				const cursor = Math.max(start, end);
 				const lastNativeText = this.component._lastNativeText || '';
-				// matches if text either starts with '/' or have (@,#,:) then it groups whatever comes next of mention type
 				const regexp = /(#|@|:|^\/)([a-z0-9._-]+)$/im;
-				const result = lastNativeText.substr(0, cursor).match(regexp);
-				if (!result) {
-					const slash = lastNativeText.match(/^\/$/); // matches only '/' in input
-					if (slash) {
-						return this.identifyMentionKeyword('', MENTIONS_TRACKING_TYPE_COMMANDS);
+				const mentionWithSearch = lastNativeText.substr(0, cursor).match(regexp);
+				if (!mentionWithSearch) {
+					const mentionWithOutSearch = lastNativeText.match(/(#|@|:|^\/)$/);
+					if (mentionWithOutSearch) {
+						return this.identifyMentionKeyword('a', mentionWithOutSearch[0]);
```

Comment:
Hello @sarthakpranesh, very thanks about your contribution.
For first, you solution not is about the issue #1580, but is another problem that we needs to resolve! :)
But, this `'a'` char isn't the better solution to this problem, how you can see [here](https://github.com/sarthakpranesh/Rocket.Chat.ReactNative/blob/Sarthak-MentionFix/app/lib/rocketchat.js#L543), we have a control to turn off promises when you're searching for something, you need to change this logic to consider a blank string in the search, and only add others mentions track on `lastNativeText.match`, it'll fix the problem.

Reply:
Cool, I'll make the required changes asap :+1: